Feature search results

nicklitvin opened the pull request:
### Changes

* Created Search Results component
* Create Stories page to showcase appearance with different props

### Usage

SearchResults contains several props for customization.

* groupBy will return the name of the group/category that a SearchResult will be placed into
* className is tailwind that is applied to the entire Accordion element
* HeaderComponent will replace the default component that renders the name of the group and can be clicked on to open/close the content
* RowComponent will replace the default component that renders each SearchResult
* ContentWrapper will replace the default wrapper around all the RowComponents in one group

The Components and Wrapper will receive props that may be useful for conditional rendering of certain parts if desired

nicklitvin committed:
added basic accordion story

nicklitvin committed:
added basic grouping of search results

nicklitvin committed:
added groupby

nicklitvin committed:
added get display

nicklitvin committed:
added dot separating and styling

nicklitvin committed:
added icon and animation

nicklitvin committed:
changed displayItem prop to JSX element

nicklitvin committed:
added collapse all and rotating icons

nicklitvin committed:
added no results option

nicklitvin committed:
fixed unique key error

nicklitvin committed:
converted getDisplay to component

nicklitvin committed:
added header component as prop

nicklitvin committed:
added more component customization to search results

nicklitvin committed:
added sorting and more props to components

nicklitvin committed:
remove reverse order prop

nicklitvin committed:
added styling to custom story

nicklitvin committed:
Merge branch 'dev' into feature-search-results

robxbob commented:
The accordian content overlaps when collapsing. Also, is there a reason why you aren't using the accordian from shadcn? Which would make your code alot simpler and not need `openCategories` and `isLast`.

Your solution of passing in components directly isn't bad at all; was just hoping for it to have a similar design structure as the model components like so:
```tsx
interface SearchResults<Category extends string> {
  data: SearchResult[];
  groupBy: (searchResult: SearchResult) => Category
  getLabel: ({ category: Category, items: SearchResult[] }) => ReactNode
  getContent: ({ category: Category, items: SearchResult[] }) => ReactNode
}
```
You can see how much more powerful the customization of the label and content can become as it now has direct access to the content and category.

robxbob commented on src/SearchResults/SearchResults.tsx:
```diff
@@ -0,0 +1,121 @@
+        <Button className="w-28" onClick={() => setOpen([])}>
+          Collapse All
+        </Button>
+      </div>
```
No need for this button

nicklitvin committed:
switched to shadcn accordion

nicklitvin committed:
Merge branch 'dev' into feature-search-results

nicklitvin committed:
refactored custom components for search results

nicklitvin committed:
added typing to search results

nicklitvin committed:
added no results customization

nicklitvin committed:
added forward ref to search results

nicklitvin committed:
fixed ref passing to search results

nicklitvin committed:
Merge branch 'dev' into feature-search-results

nicklitvin committed:
fixed linting errors

nicklitvin committed:
fixed linting errors

nicklitvin commented:
2nd round of changes:
* replaced radix accordion with shadcn accordion
* refactored SearchResults props
* added generic Category type to pass into SearchResults
* removed collpase all button 
* utilized forward ref for component
* added sortby prop

nicklitvin committed:
added search results sortby

nicklitvin committed:
fixed linting errors

robxbob commented on tailwind.config.js:
Why the sudden 4spaces tab

nicklitvin committed:
fixed tailwind config spacing

nicklitvin committed:
added space

nicklitvin committed:
removed space

nicklitvin committed:
fixed indents

robxbob commented on tailwind.config.js:
```diff
@@ -1,78 +1,86 @@
+          to: {
+            height: 'var(--radix-accordion-content-height)'
+          }
         },
```
Did you make these changes or was it something from a tailwind update lol.

nicklitvin replied:
This might have been from the install of a shadcn component that needed accordion and/or our formatter. Should I revert the change?

robxbob committed:
Merge branch 'dev' into feature-search-results
